vertx-grpc-server/src/main/java/io/vertx/grpc/server/GrpcErrorInfoProvider.java

```diff
@@ -0,0 +1,41 @@
+package io.vertx.grpc.server;
+
+import io.vertx.core.MultiMap;
+import io.vertx.grpc.common.GrpcStatus;
+
+/**
+ * Interface for providing detailed error information in gRPC server responses.
+ * <p>
+ * Implementing this interface allows exceptions to expose structured gRPC error details,
+ * including a status a descriptive error message, and optional trailers.
+ * </p>
+ * <p>
+ * This design enables custom exceptions to propagate meaningful and rich error context to gRPC clients
+ * without coupling to a specific exception class.
+ * </p>
+ */
+
+public interface GrpcErrorInfoProvider {
```

Comment:
I don't think this interface is needed for implementing the feature ? if we can just use directly `StatusException` without it, then we should do it

Reply:
Currently there are not tests using `GrpcErrorInfoProvider`, can you explain how this would be used in practice, it is not yet still clear to me how it can be used